[WEAVE] Updates tracing doc for TypeScript coverage (#1961)
wandb/docs: weave/guides/tools/attributes.mdx

</Tab>
</Tabs>

## Get attributes during execution

> You can return the current set of attributes being logged with your call. This can help you debug your calls or context for conditional logic.

Reviewer: ```suggestion
Return the current set of attributes logged to your call. This can help you debug your calls or context for conditional logic.
```

<Tabs>
<Tab title="Python">

The following example sets the Weave decorator to log the `process_data` function, configures attributes to log, and then returns them on execution.

```python lines 
import weave

weave.init("your-team/your-project")

@weave.op
def process_data(data: str):
    # Get the current call inside the op
    call = weave.get_current_call()
    if call:
        print(f"Attributes: {call.attributes}")
    return data.upper()

# Set attributes and execute the function
with weave.attributes({
    "env": "production",
    "version": "1.0.0",
    "region": "us-west-2"
}):
    process_data("hello")
    
    with weave.attributes({
        "version": "1.1.0",
        "experiment": "exp-456"
    }):
        process_data("world")
```

This outputs:

> ```

Reviewer: ```suggestion
```shell
```

Attributes: {'env': 'production', 'version': '1.0.0', 'region': 'us-west-2'}
Attributes: {'env': 'production', 'version': '1.1.0', 'region': 'us-west-2', 'experiment': 'exp-456'}
```

<Note>
`weave.get_current_call()` only works **inside** a `@weave.op` decorated function. Outside of an op, it returns `None`.
</Note>
</Tab>
<Tab title="TypeScript">

> In TypeScript, you can get the current attributes using `client.getCurrentAttributes()`. Unlike Python, this can be called from anywhere within a `withAttributes()` context, not just inside an op.

Reviewer: I was confused by this sentence. Wasn't sure what "this" was referring to in the second sentence. My attempt to make it more clear:

```suggestion
In TypeScript, you can get the current attributes using `client.getCurrentAttributes()`. Unlike Python, you can access TypeScript attributes within a `withAttributes()` context, not just inside an op.
```

Follow up question: Do you mean specifically Weave Python? A python object's attributes can be accessed anywhere within it's scope.

```typescript lines
import * as weave from 'weave';
import { withAttributes } from 'weave';

async function main() {
  const client = await weave.init('your-team/your-project');

  const processData = weave.op(async function processData(data: string) {
    // You can also access attributes inside an op
    const attrs = client.getCurrentAttributes();
    console.log('Attributes inside op:', attrs);
    return data.toUpperCase();
  });

  // Set attributes and execute the function
  await withAttributes(
    {
      env: 'production',
      version: '1.0.0',
      region: 'us-west-2'
    },
    async () => {
      // Get attributes from anywhere in the context
      console.log('Current attributes:', client.getCurrentAttributes());
      await processData('hello');

      await withAttributes(
        {
          version: '1.1.0',
          experiment: 'exp-456'
        },
        async () => {
          console.log('Nested attributes:', client.getCurrentAttributes());
          await processData('world');
        }
      );
    }
  );
}

main().catch(console.error);
```

This outputs:

> ```

Reviewer: ```suggestion
```shell
```
